darwin/importer/importer.py

```diff
@@ -71,42 +88,84 @@ def build_main_annotations_lookup_table(annotation_classes: List[Dict[str, Any]]
     return lookup
 
 
-@deprecation.deprecated(
+@deprecation.deprecated(  # type:ignore
     deprecated_in="0.7.12",
     removed_in="0.8.0",
     current_version=__version__,
     details=DEPRECATION_MESSAGE,
 )
 def find_and_parse(
-    importer: Callable[[Path], Union[List[dt.AnnotationFile], dt.AnnotationFile, None]], file_paths: List[PathLike]
+    importer: Callable[[Path], Union[List[dt.AnnotationFile], dt.AnnotationFile, None]],
+    file_paths: List[PathLike],
+    console: Optional[Console] = None,
+    use_multi_cpu: bool = True,
+    cpu_limit: int = 1,
 ) -> Optional[Iterable[dt.AnnotationFile]]:
-    # TODO: this could be done in parallel
-    for file_path in map(Path, file_paths):
-        files = file_path.glob("**/*") if file_path.is_dir() else [file_path]
-        for f in files:
-            # importer returns either None, 1 annotation file or a list of annotation files
-            parsed_files: Union[List[dt.AnnotationFile], dt.AnnotationFile, None] = importer(f)
-            if parsed_files is None:
-                continue
+    is_console = console is not None
+
+    logger = getLogger(__name__)
+
+    def perf_time(reset: bool = False) -> Generator[float, float, None]:
+        start = perf_counter()
+        yield start
+        while True:
+            if reset:
+                start = perf_counter()
+            yield perf_counter() - start
+
+    def maybe_console(*args: Union[str, int, float]) -> None:
+        if console is not None:
+            console.print(*[f"[{str(perf_time())}]", *args])
+        else:
+            logger.info(*[f"[{str(perf_time())}]", *args])
+
+    maybe_console("Parsing files... ")
+
+    files: List[Path] = _get_files_for_parsing(file_paths)
+
+    maybe_console(f"Found {len(files)} files")
+
+    if use_multi_cpu and MPIRE_AVAILABLE:
+        maybe_console(f"Using multiprocessing with {cpu_limit} workers")
+        try:
+            with WorkerPool(cpu_limit) as pool:
+                parsed_files = pool.map(importer, files, progress_bar=is_console)
+        except KeyboardInterrupt:
+            maybe_console("Keyboard interrupt. Stopping.")
+            return None
+        except Exception as e:
+            maybe_console(f"Error: {e}")
+            return None
+
+    else:
+        if use_multi_cpu and not MPIRE_AVAILABLE:
+            maybe_console("Multiprocessing module not available. Falling back to single CPU")
```

Comment:
I would add a string here like, "Using single CPU for upload. Run `pip install mpire` to benefit from faster uploads!"